docs/configfile.adoc

```diff
@@ -13,27 +13,70 @@ include::../core/src/test/resources/configuration/smart-testing.yml[]
-<8> This sets last commit sha or `HEAD` notation for inspecting changes.
-<9> This defines if smart testing should auto correct misspelled strategies.
-<10> This defines the pair key/value of custom strategies as list.
-<11> This defines list of strategy related configurations that you want to apply while applying strategy.
-<12> This enables transitivity.
-<13> This defines list of packages to be excluded while applying transitivity.
-<14> This defines list of packages to be included while applying transitivity.
+<1> Config file's absolute or relative path from where Smart Testing overwrites parameters not defined in current config file.
+<2> This defines mode to be used by Smart Testing.
+<3> This defines strategies to be used while finding important tests.
+<4> This defines plugin to be used for Smart Testing.
+<5> This enables debug logs for Smart Testing.
+<6> This disables Smart Testing if set to true.
+<7> This enables Smart Testing report if set to true.
+<8> This sets first commit sha or `HEAD` notation for inspecting changes.
+<9> This sets last commit sha or `HEAD` notation for inspecting changes.
+<10> This defines if smart testing should auto correct misspelled strategies.
+<11> This defines the pair key/value of custom strategies as list.
+<12> This defines list of strategy related configurations that you want to apply while applying strategy.
+<13> This enables transitivity.
+<14> This defines list of packages to be excluded while applying transitivity.
+<15> This defines list of packages to be included while applying transitivity.
 
 All parameters in configuration file are optional. If you haven't used any parameter in configuration file, Smart testing will use default value for that parameter.
 You can look at <<_reference_card, references>> for default value of parameter.
 
 However you can overwrite all configuration options using system properties supported by Smart Testing.
 You can look <<_reference_card, references>> for all supported system properties.
 
+=== How configuration is applying to each module?
+Smart Testing is looking in each module's root dir for configuration file. If it didn't find config file there, it'll look recursively for the first config file in
+the parent dir containing pom.xml.
+
+=== How Configuration parameters are inherited?
+If config file has `inherit` defined with absolute or relative path of config file, then Smart Testing will
+lookup for all undefined properties in config file defined using inherit. However it won't lookup for `disable` parameter.
+
+If config file doesn't contain `inherit` parameter. Then Smart Testing won't inherit any configuration properties.
+
+=== Configuration File per Maven Module:
+You can define configuration file per maven module. However it's not mandatory to define configuration per each module.
+
+NOTE: Whenever you are using configuration file per maven module, make sure to define strategies & scm properties
```

Comment:
Are scm properties really required? Why should they be?

Reply:
yes it needs to be define in parent config file OR define it using system properties. Because we are loading scm changes by looking into parent configuration file.

otherwise ST couldn't find required changes between commit ranges. as it will take default range which is `HEAD-HEAD~0`

Reply:
So it is not required but needed if you use it. Maybe we could clarify this sentence then.